contributors/design-proposals/preserve-order-in-strategic-merge-patch.md

```diff
@@ -0,0 +1,226 @@
+# Preserve Order in Strategic Merge Patch
+
+author: @mengqiy
+
+## Motivation
+
+The order of lists with a merge strategy is important, because an item in the list may depends on an item before it.
+An use case is the environment variables. We don't preserve the order which causes
+issue [40373](https://github.com/kubernetes/kubernetes/issues/40373)
+
+## Proposed Change
+
+Changes are all in strategic merge patch package.
+It will be similar to how we solve the problem of deleting items from a list of primitives.
```

Comment:
It will be similar to how we solve the problem of deleting items from a list of primitives.

> The proposed solution is similar to the solution used for deleting elements from lists of primitives.